components/server/src/migration/user-to-team-migration-service.ts

```diff
@@ -188,6 +188,14 @@ export class UserToTeamMigrationService {
     }
 
     needsMigration(user: User): boolean {
+        if (isBuiltinUser(user.id)) {
```

Comment:
From PR description:
>  This path did not take into account the "single org" constraint, that's why login/setup flows started failing on Dedicated installations.

Can we _now_ make use of the `isSingleOrgMode` config here?

Reply:
> Can we now make use of the isSingleOrgMode config here?

We could, to guard createTeam, yes. :+1: But feels like outside of scope here.

Reply:
Was more pointing to disable `needsMigration` if `isSingleOrgMode` is true.

Reply:
I think tying to `!!user.organizationId` is more accurate, as that also applies for multi-tenant.